Firestore/Source/Remote/FSTRemoteEvent.mm

```diff
@@ -435,20 +443,59 @@ - (void)addWatchChange:(FSTWatchChange *)watchChange {
   }
 }
 
+/**
+ * Updates limbo document tracking for a given target-document mapping change. If the target is a
+ * limbo target, and the change for the document has only seen limbo targets so far, and we are not
+ * already tracking a change for this document, then consider this document a limbo document update.
+ * Otherwise, ensure that we don't consider this document a limbo document. Returns true if the
+ * change still has only seen limbo resolution changes.
+ */
+- (BOOL)updateLimboDocuments:(const DocumentKey &)documentKey
+                   queryData:(FSTQueryData *)queryData
+                 isOnlyLimbo:(BOOL)isOnlyLimbo {
+  if (queryData.purpose == FSTQueryPurposeLimboResolution) {
+    // If there's already a document update for this document, it's either already in the limbo set,
```

Comment:
Did a slight variation on this. Rather than fallthrough to a final `return NO`, I inlined it in the appropriate cases so that it was more obvious which cases return `YES` vs `NO`.